client/src/RedeemPage/index.tsx

```diff
@@ -0,0 +1,156 @@
+      // const tokens = await getTokensFor(email);
+
+      // TODO: Remove next line when backend supports email
+      const mockAddress = '0x5A384227B65FA093DEC03Ec34e111Db80A040615';
+      const tokens = await getTokensFor(mockAddress);
+
+      setTokens(tokens);
+    } catch (error) {
+      addToast(error.message, {
+        appearance: 'error',
+        autoDismiss: false,
+      });
+    }
+  };
+
+  // handlers
+  const handleForm = async (values: RedeemFormValues) => {
+    setIsRedeemLoading(true);
+    const { address } = values;
+
+    redeemWithEmail(address, uid)
+      .then(() => {
+        setIsRedeemLoading(false);
+
+        addToast('POAPs loaded successfully', {
+          appearance: 'error',
+          autoDismiss: false,
+        });
+      })
+      .catch((error) => {
+        addToast(error.message, {
+          appearance: 'error',
+          autoDismiss: false,
+        });
+      });
+  };
+
+  return (
+    <div className="landing">
+      <ScanHeader sectionName="Redeem" />
+
+      <div className="redeem-content-container">
+        <Formik enableReinitialize onSubmit={handleForm} initialValues={initialValues} validationSchema={RedeemSchema}>
+          {({ dirty, isValid, handleChange, values }) => {
+            return (
+              <Form className="claim-form">
+                <span className="redeem-text">
+                  This POAPs were claimed with email <span>{email}</span>, please enter your Ethereum address or ENS to
+                  claim them
+                </span>
+                <Field
+                  name="hash"
+                  render={({ field, form }: FieldProps) => {
+                    return (
+                      <input
+                        type="text"
+                        autoComplete="off"
+                        className={classNames('width500', !!form.errors[field.name] && 'error')}
+                        placeholder="Input your Ethereum address or ENS"
+                        name="address"
+                        onChange={handleChange}
+                        value={values.address}
+                      />
+                    );
+                  }}
+                />
+                <SubmitButton
+                  className="mb-24"
+                  text="Continue"
+                  isSubmitting={isRedeemLoading}
+                  canSubmit={isValid && dirty}
+                />
+              </Form>
+            );
+          }}
+        </Formik>
+
+        <div className="redeem-text-container">
+          <span className="redeem-text">Here is the list of the POAPs to be claimed:</span>
+        </div>
+
+        <div className="redeem-poaps-container events-logos">
+          {tokens.length > 0 ? (
+            tokens.map((token) => (
+              <Link
+                key={token.tokenId}
+                to={{
+                  pathname: `/token/${token.tokenId}`,
+                  state: token,
```

Comment:
Why do we send or use router state?

Reply:
This is my bad for not checking if was needed or not, but it's handled in the component, ergo, considered a possible flow of data. I attach a image showing that.

<img width="763" alt="Screen Shot 2020-07-27 at 10 58 06" src="https://user-images.githubusercontent.com/8381041/88550491-0f138d00-cff8-11ea-83a7-af23042a5bd0.png">